internal/server/iigointernal/base_speaker.go

```diff
@@ -0,0 +1,172 @@
+package iigointernal
+
+import (
+	"math/rand"
+	"time"
+
+	"github.com/SOMAS2020/SOMAS2020/internal/common/gamestate"
+	"github.com/SOMAS2020/SOMAS2020/internal/common/roles"
+	"github.com/SOMAS2020/SOMAS2020/internal/common/rules"
+	"github.com/pkg/errors"
+)
+
+type baseSpeaker struct {
+	id            int
+	budget        int
+	judgeSalary   int
+	ruleToVote    string
+	votingResult  bool
+	clientSpeaker roles.Speaker
+}
+
+func (s *baseSpeaker) withdrawJudgeSalary(gameState *gamestate.GameState) error {
+	var judgeSalary = int(rules.VariableMap["judgeSalary"].Values[0])
+	var withdrawError = WithdrawFromCommonPool(judgeSalary, gameState)
+	if withdrawError != nil {
+		featureSpeaker.judgeSalary = judgeSalary
+	}
+	return withdrawError
+}
+
+func (s *baseSpeaker) sendJudgeSalary() {
+	if s.clientSpeaker != nil {
+		amount, err := s.clientSpeaker.PayJudge()
+		if err == nil {
+			featureJudge.budget = amount
+			return
+		}
+	}
+	amount, _ := s.payJudge()
+	featureJudge.budget = amount
+}
+
+// Pay the judge
+func (s *baseSpeaker) payJudge() (int, error) {
+	hold := s.judgeSalary
+	s.judgeSalary = 0
+	return hold, nil
+}
+
+// Receive a rule to call a vote on
+func (s *baseSpeaker) setRuleToVote(r string) {
+	s.ruleToVote = r
+}
+
+//Asks islands to vote on a rule
+//Called by orchestration
+func (s *baseSpeaker) setVotingResult() {
+	if s.clientSpeaker != nil {
+		result, err := s.clientSpeaker.RunVote(s.ruleToVote)
+		if err != nil {
+			s.votingResult, _ = s.runVote(s.ruleToVote)
```

Comment:
Same as other error comments, if runVote never returns any error other than nil, then you should just remove it from the interface